crates/bitwarden-core/src/key_management/user_decryption.rs

```diff
@@ -0,0 +1,142 @@
+#![allow(missing_docs)]
+
+use bitwarden_api_api::models::UserDecryptionResponseModel;
+use bitwarden_error::bitwarden_error;
+use serde::{Deserialize, Serialize};
+
+use crate::key_management::master_password::{MasterPasswordError, MasterPasswordUnlockData};
+
+#[bitwarden_error(flat)]
+#[derive(Debug, thiserror::Error)]
+pub enum UserDecryptionError {
+    #[error(transparent)]
+    MasterPasswordError(#[from] MasterPasswordError),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[cfg_attr(
+    feature = "wasm",
+    derive(tsify::Tsify),
+    tsify(into_wasm_abi, from_wasm_abi)
+)]
+pub struct UserDecryptionData {
+    pub master_password_unlock: Option<MasterPasswordUnlockData>,
+}
+
+impl UserDecryptionData {
+    pub fn process_response(
+        response: UserDecryptionResponseModel,
+    ) -> Result<UserDecryptionData, UserDecryptionError> {
+        let master_password_unlock = match response.master_password_unlock {
+            Some(master_password_unlock) => Some(MasterPasswordUnlockData::process_response(
+                master_password_unlock.as_ref().clone(),
+            )?),
+            None => None,
+        };
```

Comment:
```suggestion
        let master_password_unlock = response
            .master_password_unlock
            .map(|mpu| MasterPasswordUnlockData::process_response(*mpu))
            .transpose()?;
```
(or the try from equivalent)

Reply:
Fixed in 70ad9d37